### DISCUSSION TITLE
How can I set bandwidth limit in WebRTC?

### DISCUSSION THREAD

**Discussion by sevkqo** [sipsorcery-org/sipsorcery#1124] [Q&A]
Hi,
is it possible to set bandwidth limit in WebRTC? I've tried to modify SDP file by adding "b=TIAS:..." parameter:

`b=TIAS:100000` in ffmpeg.sdp file
`offerInit.sdp = offerInit.sdp.Replace("c=IN IP4 0.0.0.0", "c=IN IP4 0.0.0.0\nb=TIAS:300000");`

and set parameter "MaximumBandwidth" to an instance of MediaStreamTrack:

`MediaStreamTrack videoTrack = new MediaStreamTrack(SDPMediaTypesEnum.video, false, new List<SDPAudioVideoMediaFormat> { videoFormat }, MediaStreamStatusEnum.SendRecv) {Ssrc = SSRC_REMOTE_VIDEO, MaximumBandwidth = 75000}`

, but it did not change anything. I'm streaming video in 1080p, framerate is stable (~30 fps) and calculated bitrate is about 1 MB/s, so i expected that WebRTC change video resolution when i set bandwidth limitation.
I'm currently working on a "FfmpegToWebRTC" example.
Thanks.

**Comment by ha-ves** [comment]
FFmpeg bitrate setting is available by PR sipsorcery-org/SIPSorceryMedia.FFmpeg#62.

However, no WebRTC `ontrack` event yet and no TIASBandwidth usage internally.

Application workaround is maybe to subscribe on offer-answer signaling (after `setRemoteDescription`) and poll the MediaStreamTrack(s) for the MaxBandwidthValue and set FFmpegVideoSource's encoder bitrate accordingly.
